### PR TITLE
Use all kops mirrors to determine artifacts hashes

### PR DESCRIPTION
**Pull request by hakman**
Current implementation only looks for hash files in the S3 mirror and, if that is down, kops commands will fail with:
```
cannot determine hash for "https://kubeupv2.s3.amazonaws.com/kops/1.19.0-alpha.4/linux/amd64/nodeup"
(have you specified a valid file location?)
```

The code in the assets remapping and mirroring is quite complex and even if this looks a bit less elegant than I would like, it works pretty well.

Fixes: #9951

### CODE REVIEW

**Review by zetaab — APPROVED** [review]
/lgtm

**Comment by rifelpet** [comment]
This new logic and the constants look very similar to https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/urls.go would it make sense to consolidate them?

**Comment by hakman** [comment]
> This new logic and the constants look very similar to https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloudup/urls.go would it make sense to consolidate them?

It would make sense to move this to a function somewhere they both could use, but no idea where. It can also be done in a followup PR once we agree where to extract it.

File: pkg/assets/builder.go [hunk 1]
```diff
@@ -311,27 +312,30 @@ func (a *AssetBuilder) findHash(file *FileAsset) (*hashing.Hash, error) {
...
-				} else {
-					klog.V(2).Infof("unable to read hash file %q: %v", hashURL, err)
+			for _, mirror := range mirrors.FindUrlMirrors(u.String()) {
+				hashURL := mirror + ext
+				klog.V(3).Infof("Trying to read hash fie: %q", hashURL)
+				b, err := vfs.Context.ReadFile(hashURL, vfs.WithBackoff(backoff))
+				if err != nil {
+					// Try to log without being too alarming - issue #7550
+					if ext == ".sha256" {
+						klog.V(2).Infof("Unable to read new sha256 hash file (is this an older/unsupported kubernetes release?) %q: %v", hashURL, err)
```
Comment (johngmyers) [inline pkg/assets/builder.go:322]: Should log this only after trying all mirrors.

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: pkg/assets/builder.go (hunk 1 above)
Comment (hakman) [reply pkg/assets/builder.go]: Good idea. Done.

**Review by zetaab — APPROVED** [review]
/lgtm

**Pull request merged**